### PR TITLE
spanner-jdbc-cleanup - Step 11 - Add integration tests based on music data model

### PR DESCRIPTION
**Pull request by olavloite**
Adds SQL based integration tests using the standard music data model.

### CODE REVIEW

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/it/ITSqlMusicScriptTest.java [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    final long VENUE_ID = 68L;
+    long numberOfSongs = 0L;
+    AbortInterceptor interceptor = new AbortInterceptor(0.0D);
+    try (ITConnection connection = createConnection(interceptor)) {
+      connection.setAutocommit(false);
+      connection.setRetryAbortsInternally(true);
+      // read all data from the different music tables in the transaction
+      try (ResultSet rs = connection.executeQuery(Statement.of("SELECT * FROM Singers"))) {
+        while (rs.next()) {
+          // do nothing
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/it/ITSqlMusicScriptTest.java]: let's add some basic assertions here on the result set? Can be simple like asserting the id, name, etc

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/resources/com/google/cloud/spanner/jdbc/ITSqlMusicScriptTest.sql [hunk 2]
```diff
@@ -0,0 +1,670 @@
...
...oJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBrc96XvRx/YfuhN0f0nr7w+ur/1D34W3X/3zx+I7h885cXo/p3XDET3T/3UMdH9+9ZOje5fvOXy6P7dt384uj/tBwdG92fNeDu6v2veG9H9M762Iro/4bjHovsuAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBIAgFICAFBKAABKCQBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUEoAAEoJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACg1MCIhf8XfeAP/2s4uv/crr+L7k895J7o/oxDbo7uf3fm0uj+uaN+Et1fMfDD6P4FU7J//8b1b0f3D/3mQ9H90RecHt0f89G9o/vPf2lJdP/ji3dG9289OLvvAgAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBIAgFICAFBKAABKCQBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUEoAAEoJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASg186qzx0Qfe2bFfdP/B9306uv/5954T3f/iTx+O7l/yyKXR/YETn4ruH/1Pc6P7/3zyFdH9Px15RHR/8XuPje4PrZwd3T/t7KXR/f+dvCq6f9VbL0b33z3+19F9FwBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUEoAAEoJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBIAgFICAFBKAABKCQBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUGpw4iUfiz6wYtXY6P7s7zwf3d9xxZLo/titM6P7mz+R3Z8246Lo/jur1kf3D56+Mrp//u7d0f2Fd94f3b//6zui+xsn3xzdv3bH6dH9cXuOiO7f+p8jo/suAIBSAgBQSgAASgkAQCkBACglAAClBACglAAAlBIAgFICAFBKAABKCQBAKQEAKCUAAKUEAKCUAACUEgCAUgIAUEoAAEoJAEApAQAoJQAApQQAoJQAAJQSAIBSAgBQSgAASgkAQCkBACglAAClBACg1P8HAAD//+SLeeht2MttAAAAAElFTkSuQmCC'),DATE '1906-04-28');
```
Comment (kolea2) [inline google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/resources/com/google/cloud/spanner/jdbc/ITSqlMusicScriptTest.sql]: what is the FROM_BASE64 stuff?

Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/resources/com/google/cloud/spanner/jdbc/ITSqlMusicScriptTest.sql]: Randomly generated PNG images encoded as BASE64 to test working with byte arrays. Copy-paste a value here to have a look: https://onlinepngtools.com/convert-base64-to-png

File: google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/it/ITSqlMusicScriptTest.java (hunk 1 above)
Comment (olavloite) [reply google-cloud-clients/google-cloud-contrib/google-cloud-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/it/ITSqlMusicScriptTest.java]: Added assertions for the ID's and number of records that are read from the database. The first test script adds but also deletes some of the records, so the assertions contain some if-statements checking for specific ID's that it knows have been deleted.

**Comment by olavloite** [comment]
Closing in favor of new PR based on master.

**Pull request closed without merging**